File: ios/Classes/Models/PowerMode.swift
import Foundation

/// Power mode for battery optimization
enum PowerMode: Int {
    case performance = 0
    case balanced = 1
    case powerSaver = 2
    case ultraLowPower = 3

    static func fromInt(_ value: Int) -> PowerMode {
        return PowerMode(rawValue: value) ?? .balanced
    }
}

File: android/src/main/kotlin/com/ble_mesh/models/PowerMode.kt
package com.ble_mesh.models

/**
 * Power mode for battery optimization
 */
enum class PowerMode(val value: Int) {
    PERFORMANCE(0),
    BALANCED(1),
    POWER_SAVER(2),
    ULTRA_LOW_POWER(3);

    companion object {
        fun fromInt(value: Int) = values().firstOrNull { it.value == value } ?: BALANCED
    }
}

